pkgs/development/web/bun/default.nix

```diff
@@ -0,0 +1,59 @@
+{ stdenvNoCC, callPackage, fetchurl, autoPatchelfHook, unzip, openssl, lib }:
+let
+  dist = {
+    aarch64-darwin = {
+      arch = "aarch64";
+      shortName = "darwin";
+      sha256 = "06kf4dyb8y486r2bbpz8znnaxah9y2hwc9km3cigacfchi4dy28j";
+    };
+
+    x86_64-darwin = {
+      arch = "x64";
+      shortName = "darwin";
+      sha256 = "06q8cajxra8lgyd1ap6iqzqdw31gkvlc6l2ma0mm22y63msgdn8b";
+    };
+
+    x86_64-linux = {
+      arch = "x64";
+      shortName = "linux";
+      sha256 = "038w54yyk94042hg27hfm49w12xsg8wh0skza9l9iybvc9qlvh2i";
+    };
+  }.${stdenvNoCC.hostPlatform.system} or (throw "Unsupported system: ${stdenvNoCC.hostPlatform.system}");
+  version = "0.1.1";
+  pname = "bun";
+  meta = with lib; {
+    homepage = "https://bun.sh";
+    changelog = "https://github.com/Jarred-Sumner/bun/releases/tag/bun-v${version}";
+    description = "Incredibly fast JavaScript runtime, bundler, transpiler and package manager – all in one";
+    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
+    longDescription = ''
+      All in one fast & easy-to-use tool. Instead of 1,000 node_modules for development, you only need bun.
+    '';
+    license = licenses.mit;
+    maintainers = with maintainers; [ DAlperin jk ];
+    platforms = [ "x86_64-linux" "x86_64-darwin" "aarch64-darwin" ];
```

Comment:
If you separate the dists attribute set (object)

E.g.

dists = { x86-64_linux: { ... }, ... }
dist = dists.${xyz}

You can then build the platforms list like so:

`platforms = builtins.attrNames dists;`

https://nixos.org/manual/nix/stable/expressions/builtins.html#builtins-attrNames

And it'll always be up to date

Reply:
Good idea, thank you